services/libs/integrations/src/integrations/github/processWebhookStream.ts

```diff
@@ -0,0 +1,361 @@
+
+  await ctx.publishData<GithubWebhookData>({
+    webhookType: GithubWehookEvent.DISCUSSION,
+    subType: GithubWebhookSubType.DISCUSSION_COMMENT_START,
+    data: payload,
+    member,
+  })
+}
+
+const parseWebhookPullRequestEvents = async (
+  payload: any,
+  ctx: IProcessWebhookStreamContext,
+): Promise<void> => {
+  const member = await prepareWebhookMember(payload.sender.login, ctx)
+  let objectMember: GithubPrepareMemberOutput | undefined
+
+  const GITHUB_CONFIG = ctx.platformSettings as GithubPlatformSettings
+  const IS_GITHUB_COMMIT_DATA_ENABLED = GITHUB_CONFIG.isCommitDataEnabled === 'true'
+
+  switch (payload.action) {
+    case 'edited':
+    case 'opened':
+    case 'reopened':
+    case 'closed':
+    case 'merged': {
+      await ctx.publishData<GithubWebhookData>({
+        webhookType: GithubWehookEvent.PULL_REQUEST,
+        data: payload,
+        member,
+      })
+      break
+    }
+    case 'assigned':
+    case 'review_requested': {
+      objectMember = await prepareWebhookMember(payload.requested_reviewer.login, ctx)
+
+      await ctx.publishData<GithubWebhookData>({
+        webhookType: GithubWehookEvent.PULL_REQUEST,
+        data: payload,
+        member,
+        objectMember,
+      })
+      break
+    }
+    case 'synchronize': {
+      if (IS_GITHUB_COMMIT_DATA_ENABLED) {
+        const prNumber = payload.number
+        const repo: Repo = {
+          name: payload.repository.name,
+          owner: payload.repository.owner.login,
+          url: payload.repository.html_url,
+          createdAt: payload.repository.created_at,
+        }
+
+        // this will create a CROWD_GENERATED webhook and stream for it
+        // this way we don't need integration run to publish new streams
+        await ctx.publishStream<GithubBasicStream>(
+          `${GithubStreamType.PULL_COMMITS}:${prNumber}:firstPage`,
+          {
+            repo,
+            page: '',
+            prNumber,
+          },
+        )
+      }
+      break
+    }
+  }
+}
+
+const parseWebhookPullRequest = async (payload: any, ctx: IProcessWebhookStreamContext) => {
+  // handle case of multiple reviewers (by assigning a team as a reviewer)
+  if (payload.action === 'review_requested' && payload.requested_team) {
+    // a team sent as reviewer, first we need to find members in this team
+    const team: GithubWebhookTeam = payload.requested_team
+    const teamMembers = await new TeamsQuery(team.node_id, ctx.integration.token).getSinglePage('')
+
+    for (const teamMember of teamMembers.data) {
+      await parseWebhookPullRequestEvents({ ...payload, requested_reviewer: teamMember }, ctx)
+    }
+  }
+
+  if (payload.action === 'closed' && payload.pull_request.merged) {
+    const revisedPayload = { ...payload, action: 'merged' }
+    revisedPayload.pull_request.state = 'merged'
+
+    await parseWebhookPullRequestEvents(revisedPayload, ctx)
+  }
+
+  await parseWebhookPullRequestEvents(payload, ctx)
+}
```

Comment:
In the current code we're breaking the switch statement in the ifs, but here we always check other ifs and always call the last one. I think we should return instead in if blocks so the last one that's not in ifs won't get published unnecessarily

Reply:
Yeah I agree, this is actually a bug